### PR TITLE
chore: nuke isPage, fix bugs

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

**Review by dgozman — APPROVED** [review]

File: src/Playwright.Tests/PageNetworkResponseTests.cs [hunk 1]
```diff
@@ -63,7 +64,9 @@ public async Task ShouldReturnJson()
         public async Task ShouldWorkWithGenerics()
         {
             var response = await Page.GotoAsync(TestConstants.ServerUrl + "/simple.json");
-            Assert.Equal("bar", (await response.JsonAsync<TestClass>()).Foo);
+            JsonDocument doc = await response.JsonAsync();
```
Comment (kblok) [inline src/Playwright.Tests/PageNetworkResponseTests.cs:67]: The generic JsonAsync would be way more useful than this. I don't think people would use this with all these required boilerplate.

File: src/Playwright.Tests/PageNetworkResponseTests.cs [hunk 2]
```diff
@@ -159,7 +162,7 @@ public async Task ShouldWaitUntilResponseCompletes()
 
         class TestClass
```
Comment (kblok) [inline src/Playwright.Tests/PageNetworkResponseTests.cs:163]: We can drop this class if we are not going to use it for tests.

File: src/Playwright/BrowserContext.cs [hunk 3]
```diff
@@ -203,13 +203,14 @@ public Task SetExtraHTTPHeadersAsync(IEnumerable<KeyValuePair<string, string>> h
...
+            var options = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
             string state = JsonSerializer.Serialize(
                 await Channel.GetStorageStateAsync().ConfigureAwait(false),
-                Channel.Connection.GetDefaultJsonSerializerOptions());
+                options);
 
             if (!string.IsNullOrEmpty(path))
             {
-                File.WriteAllText(path, state);
+                await File.WriteAllTextAsync(path, state).ConfigureAwait(false);
```
Comment (kblok) [inline src/Playwright/BrowserContext.cs:213]: I think WriteAllTextAsync is net 5 only.

**Review by dgozman — APPROVED** [review]

File: src/Playwright.Tests/PageNetworkResponseTests.cs (hunk 1 above)
Comment (pavelfeldman) [reply src/Playwright.Tests/PageNetworkResponseTests.cs:67]: > The generic JsonAsync would be way more useful than this. 

I don't this force-converting to camel case is going to have a single use, default one can at least be used...

File: src/Playwright/BrowserContext.cs (hunk 3 above)
Comment (pavelfeldman) [reply src/Playwright/BrowserContext.cs]: Yup, figured that!

File: src/Playwright.Tests/PageNetworkResponseTests.cs (hunk 2 above)
Comment (pavelfeldman) [reply src/Playwright.Tests/PageNetworkResponseTests.cs:163]: Will follow-up

**Pull request merged**